doc/releases/changelog-dev.md

```diff
@@ -87,6 +87,14 @@
 * When given a callable, `qml.ctrl` now does its custom pre-processing on all queued operators from the callable.
   [(#4370)](https://github.com/PennyLaneAI/pennylane/pull/4370)
 
+* `qml.interfaces.set_shots` can now accepts shots specified as a `qml.measurements.Shots` object.
+  If a `pennylane.devices.experimental.Device` is provided, the `set_shots` will no longer raise an error, but will simply return without doing anything.
```

Comment:
This makes a lot of internal code eaiser, like for the original code:
```
                if isinstance(self._original_device, Device):
                     set_shots(self._original_device, override_shots)(self._update_gradient_fn)()
                 else:
                     self._update_gradient_fn(shots=override_shots)
```
We can reduce it to only:
```
                     set_shots(self._original_device, override_shots)(self._update_gradient_fn)()
```
by moving that `isinstance` check inside `set_shots`.

Reply:
I guess it comes down to whether or not `set_shots` is an internal tool or user facing.

Reply:
Ah... @trbromley this is worth double checking/considering. See if people are using `set_shots`, also see if it's been mentioned on github issues, in the forum, etc.

I have definitely been using it.

Reply:
Since both Tom and Matthew had concerned, it now raises an explicit `ValueError` with the new device.  Makes some code a little more branch-y, but I'm ok with that.

Reply:
@albi3ro you're still doing the `if-else` [on this branch](https://github.com/PennyLaneAI/pennylane/blob/shots-on-new-device/pennylane/qnode.py#L944-L949), is that intentional/necessary?